odoo/cli/load_db.py

```diff
@@ -0,0 +1,69 @@
+import argparse
+import io
+import urllib.parse
+from pathlib import Path
+from sys import argv
+
+import requests
+
+from . import Command
+from .server import report_configuration
+from ..service.db import restore_db
+from ..tools import config
+
+
+class LoadDb(Command):
+    name = 'load_db'
+
+    def run(self, cmdargs):
+        """Loads an Odoo dump into the current system.
+
+        Aside from ``--database``, the options match (and are directly forwarded
+        too) their regular Odoo behaviour.
+        """
+        parser = argparse.ArgumentParser(
+            prog="%s %s" % (Path(argv[0]).name, self.name),
+            description=self.__doc__
+        )
+        parser.add_argument(
+            '-d', '--database',
+            help="Database to restore to, if not provided uses the dump file's"
+                 " filename (without extension)"
+        )
+        parser.add_argument('-c', '--config')
+        parser.add_argument('-D', '--data-dir')
+        parser.add_argument('--addons-path')
+        parser.add_argument('-r', '--db_user')
+        parser.add_argument('-w', '--db_password')
+        parser.add_argument('--pg_path')
+        parser.add_argument('--db_host')
+        parser.add_argument('--db_port')
+        parser.add_argument('--db_sslmode')
+        parser.add_argument('dump_file')
+        args = parser.parse_args(cmdargs)
+
+        config.parse_config([
+            val
+            for k, v in vars(args).items()
+            if v is not None
+            if k != 'dump_file'
+            for val in [k, v]
+        ])
```

Comment:
So this bit is to reconstruct a sys.argv like list without the positional `dump_file` am I right?

So if we do `odoo-bin load_db --db_host localhost -c config.cfg path/to/my/dump.sql` doesnt' this list comprehension gives `['db_host', 'localhost', 'config', 'config.cfg']`, aren't the dashes missing for db_host and config?

Shouldn't it be `for var in [f'--{k}', v]`?

Reply:
> Shouldn't it be `for var in [f'--{k}', v]`?

Yep, kinda forgot to test that... also needs to revert the key back to the CLI form, as argparse will convert the dashes to underscores...

Reply:
For what it's worth

    parser = ArgumentParser()
    next((action.option_strings[0] for action in parser._actions if action.dest == 'help'))
    '-h'

Reply:
Yeah I know. But I can probably just have a map for the exceptions and return the others as-is.